app/views/collection_items/_item_fields.html.erb

```diff
@@ -7,9 +7,13 @@
   <%= fields_for("collection_items[]", collection_item) do |form| %>
     <!--title-->
     <h4 class="heading" id="collection_item_<%= collection_item.id %>">
-      <%= error_messages_for collection_item %>    
+      <%= error_messages_for collection_item %>
       <%= form.hidden_field :collection_id %>
-      <%= link_to(item_name, collection_item.item) %>   
+      <% if collection_item.item_type == "Work" && collection_item.item.in_unrevealed_collection? && !collection_item.collection.unrevealed? %>
```

Comment:
"Mystery work" should appear to moderators (and participants) of Revealed collections, so long as the Work is also currently a part of an Unrevealed collection. The idea (might not have been intentional) is that moderators of a Revealed collection could theoretically also be a part (as participants) of the Unrevealed collection, so being able to see the name of the Work might spoil it for them. 

But like I said, if they wanted to be sneaky then switching their collection back to 'Unrevealed' would show them the Work details. This is most likely a fringe case, but it could be used abusively.